site/src/_pages/spec/2_embedding-applications.mdx

```diff
@@ -28,20 +45,24 @@ An embedding application SHOULD be able to read a JSON schema file provided by a
 
 ## Styling
 
-Embedding applications SHOULD define, in the global scope, any of the CSS variables listed in [Appendix A](https://blockprotocol.org/spec/appendix-a-styling) which they have a preferred value for, and additionally provide these values via a `styleVariables` object passed to the block alongside other data.
+Embedding applications SHOULD provide a `styleVariables` object passed to the block alongside other fields, containing any of the theme variables listed in [Appendix A](https://blockprotocol.org/spec/appendix-a-styling) which they have a preferred value for.
+
+Blocks SHOULD use these variables where appropriate to make their block visually consistent with the embedding application (except where applying application-supplied values may interfere with block functionality or accessibility).
 
 ## Security
 
 Naively running third-party code is inherently unsafe.
-Embedding applications should take suitable precautions to ensure that block code they execute is unable to compromise their systems or their users’ data.
+
+Embedding applications SHOULD take suitable precautions to ensure that block code they execute is unable to compromise their systems or their users’ data.
 Risk mitigation might include various approaches to sandbox block code execution.
 
-Rendering HTML blocks inside iframes or web workers, and transferring data and offering function calls via messages, are obvious mitigations to explore.
+Rendering HTML blocks inside iframes or web workers are obvious mitigations to explore.
+
+In these types of implementations, the embedding application would run a small application inside the iframe or web worker. This sandbox application would in turn render the block, and provide it with the functions and data it needs.
+
+When the block calls a function (e.g. updateEntity), the sandboxing application sends messages with the request and payload to the parent application. An example iframe implementation can be found in [HASH, a working demo for a block-embedding application](https://github.com/hashintel/hash).
 
-In these types of implementations, the embedding application would render a small application inside of an iframe or web worker.
-This sandbox application would in turn render the block, and provide it with the functions and data it needs.
-When the block calls a function (e.g. updateEntity), the sandboxing application sends messages with the request and payload to the parent application.
-Users should then be prompted to permit or deny the sort of action the block is attempting to take, and their preferences saved.
+Users SHOULD then be prompted to permit or deny the sort of action the block is attempting to take, and their preferences saved. The permissions a block requires could also be [sought and approved in advance when a block is loaded](https://blockprotocol.org/block-types#block-permissions).
```

Comment:
```suggestion
Users COULD then be prompted to permit or deny the sort of action the block is attempting to take, and their preferences saved. The permissions a block requires could also be [sought and approved in advance when a block is loaded](https://blockprotocol.org/block-types#block-permissions).
```

I think whether this is a 'SHOULD' depends on the environment the application is being hosted in and the breadth of blocks that are being utilized (e.g. full catalog inc. untrusted blocks versus a subset). Would be tempted to make this 'COULD' to highlight it's a recommend viable option depending on a user's scenario, versus a 'SHOULD' which may imply it's more globally best-practice.